projects/packages/sync/src/modules/class-woocommerce-products.php

```diff
@@ -353,4 +361,26 @@ public function get_next_chunk( $config, $status, $chunk_size ) {
 			'objects'    => $filtered_product_data,
 		);
 	}
+
+	/**
+	 * Convert the WC_DateTime objects to stdClass objects to ensure they are properly encoded.
+	 *
+	 * @param WC_DateTime|mixed $wc_datetime The datetime object.
+	 * @param bool              $utc         Whether to convert to UTC.
+	 * @return object|null
+	 */
+	private function datetime_to_object( $wc_datetime, $utc = false ) {
+		if ( is_string( $wc_datetime ) ) {
+			$wc_datetime = new WC_DateTime( $wc_datetime, new DateTimeZone( wc_timezone_string() ) );
+		}
+
+		if ( is_a( $wc_datetime, 'WC_DateTime' ) ) {
+			if ( $utc ) {
+				$wc_datetime->setTimezone( new DateTimeZone( 'UTC' ) );
+			} else {
+				$wc_datetime->setTimezone( new DateTimeZone( wc_timezone_string() ) );
+			}
+			return (object) (array) $wc_datetime;
+		}
+	}
```

Comment:
```suggestion
		}
		
		return null;
	}
```

Nitpick, feel free to ignore: Adding an explicit return statement.